### ISSUE TITLE
Conditional checks for Composition API

### ISSUE THREAD

**Issue by assassin316** [Microsoft/WindowsUIDevLabs#81]
Is there a simple example that shows how we can have conditional checks on the composition API features?

For example, suppose I want to add blur in an app, BUT the user has a windows version that includes older Composition features, not the latest ones from BUILD. How can I protect the code so that newer features would show on newer versions of Windows only?

Thanks!

**Comment by pgills** [comment]
@assassin316 A the moment we don't have a sample that demonstrates how to use API checks for fallback when using the API. However, we are hearing that feedback from many customers using the API that they would like a more consumable list of what's supported on the major builds.

One technique you can use is the `ApiInformation` API to determine a bunch of things at runtime:
- If a certain API contract is present - `IsApiContractPresent`
- If a certain Property is present - `IsPropertyPresent` - can be helpful for determining the `Shadow` property is present on a `Visual`.

For more reference: https://msdn.microsoft.com/en-us/library/windows/apps/windows.foundation.metadata.apiinformation.aspx

As you pointed out, the API has some more subtle differences where you'll be able to specify the `GaussianBlurEffect`, but without knowing which build you are on, the effect might not be supported.

The key takeaways for us:
- Provide a simple sample demonstrating using `ApiInformation` APIs
- Provide a reference of subtle build to build differences that cannot be detected at runtime using the `ApiInformation` APIs

Here's an article that provides some basics on feature detection: https://blogs.windows.com/buildingapps/2015/09/15/dynamically-detecting-features-with-api-contracts-10-by-10/

**Comment by assassin316** [comment]
Thanks for the info! 

Would these be correct ??

 		// SDK 10240 and later. 
		if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 1))
		{
			...
		}

		// SDK 10586 and later.
		if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 2))
		{
			...
		}

		// SDK 14332 and later.
		if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 3))
		{
			...
		}

In other words, using a Blur animation, for example, we would check against 14332 and later.

**Comment by jeffstall** [comment]
Yes—this is correct.
